inc/sitemaps/class-author-sitemap-provider.php

```diff
@@ -0,0 +1,262 @@
+<?php
+/**
+ * @package WPSEO\XML_Sitemaps
+ */
+
+/**
+ * Sitemap provider for author archives.
+ */
+class WPSEO_Author_Sitemap_Provider implements WPSEO_Sitemap_Provider {
+
+	/**
+	 * Set up filter for excluded authors.
+	 */
+	public function __construct() {
+
+		add_filter( 'wpseo_sitemap_exclude_author', array( $this, 'user_sitemap_remove_excluded_authors' ), 8 );
+	}
+
+	/**
+	 * Check if provider supports given item type.
+	 *
+	 * @param string $type Type string to check for.
+	 *
+	 * @return boolean
+	 */
+	public function handles_type( $type ) {
+
+		return $type === 'author';
+	}
+
+	/**
+	 * @param int $max_entries Entries per sitemap.
+	 *
+	 * @return array
+	 */
+	public function get_index_links( $max_entries) {
```

Comment:
This method has multiple responsibilities, maybe it worth considering these in separate methods.